### PR TITLE
Added requirments, changelog, readme and version files, optimized code

### PR DESCRIPTION
**Pull request by tommybrecher**
Hi @AlexSpodarev ,

Would appreciated your humble opinion on the suggested code I want to push to our project:

In short, I added the following:

- version.txt file - (we will need to update it every time we want to merge a new version to master, currently we are in 0.0.1 if you approve this pull request we will go to 0.0.2, that's why I put it on 0.0.2

- CHANGELOG.md file - This is were we put the changes that were done when incrementing the version, see my examples.

- requirments.txt file - This is for users who want to clone the repository and need to know which python packages to have installed.

- README.md file - Just a simple getting started guide, how to install how to run , etc

- REST_API.py - Well... look at the CHANGELOG.md file and try to figure out what features I tried to achieve

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+
+## [0.0.2] - 2018-12-25
+
+### Added
+- CHANGELOG.md file to keep track of changes - [@tommybrecher](https://github.com/tommybrecher)
+- requirments.txt so users who clone the repository can run `pip install -r requriments.txt` - [@tommybrecher](https://github.com/tommybrecher) 
+- Comments on chunks of the code where the logic gets complex - [@tommybrecher](https://github.com/tommybrecher)
+- A method which aborts invalid requests with a 404 unless a specific code is provided `reject_invalid_request()` - [@tommybrecher](https://github.com/tommybrecher)
+- logging statments everywhere for easier debugging (shouldn't run anyways if when calling app.run() we don't set it to debugging - [@tommybrecher](https://github.com/tommybrecher)
```
Comment (AlexSpodarev) [inline CHANGELOG.md]: *statements

File: README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+- [@tommybrecher](https://github.com/tommybrecher)
+
+If you wish to contribute, go ahead!
+
+
+### Getting started
+
+
+1) Create a virtualenv or use your local setup of Python3
+2) Install the requirments with pip (pip3 if you have both python2/3 installed) as following pip install -r requirements.txt
```
[the remark this answers is not in the document]
Comment (tommybrecher) [reply README.md]: Resolved

File: CHANGELOG.md (hunk 1 above)
Comment (tommybrecher) [reply CHANGELOG.md]: Resolved

**Comment by tommybrecher** [comment]
@AlexSpodarev - Addressed the points you brought up, thanks it was very informative :)

Anything else?

**Pull request merged**